### PR TITLE
winch: Switch to using cranelift for all trampolines

### LINKED ISSUE #8098: winch: Use cranelift as the trampoline compiler
https://github.com/bytecodealliance/wasmtime/pull/8082 introduces Cranelift as the trampoline compiler for components in Winch. This allows reducing code duplication and allowing for better maintainability. 

In order to have uniform trampoline compilation across the board, we want to migrate the rest of Winch's trampolines to use Cranelift.

### PR DESCRIPTION
**Pull request by elliottt**
Remove the winch MacroAssembler-based trampolines, and reuse the trampoline-generation from the `wasmtime-cranlift` crate instead.

Fixes #8098

### CODE REVIEW

**Review by fitzgen — APPROVED** [review]
Nice!

File: crates/cranelift/src/lib.rs [hunk 1]
```diff
@@ -154,7 +154,7 @@ fn wasm_call_signature(
         }
 
         // The winch calling convention is only implemented for x64 and aarch64
-        arch if tunables.tail_callable => {
+        arch if tunables.winch_callable => {
```
Comment (elliottt) [inline crates/cranelift/src/lib.rs:157]: If this PR gets reverted, please preserve this change.

**Review by saulecabrera — APPROVED** [review]
LGTM, thanks!

One thing that I wanted to note: Winch is still missing support for `emit_trampolines_for_array_call_host_func`, would it make sense to implement that function in this PR? I think a follow-up also works!

File: crates/cranelift/src/lib.rs (hunk 1 above)
Comment (elliottt) [reply crates/cranelift/src/lib.rs]: I merged this separately, so we can ignore this comment now.

**Pull request merged**